runtime/common/QuditIdTracker.h

```diff
@@ -36,15 +36,15 @@ class QuditIdTracker {
     }
 
     auto next = recycledQudits.front();
-    recycledQudits.pop_front();
+    recycledQudits.erase(recycledQudits.begin());
```

Comment:
If you sort the indices in descending order (as bellow), this becomes:
```cpp
auto next = recycledQudits.back();
recycledQudits.pop_back();
```
(which will be slightly better than removing from the front and shifting the rest of the vector---might not make a big difference depending on how big you expect this array to be)

Out of curiosity, what is the advantage of getting the lowest index among the set of previously returned indices ?

Reply:
I don't think there really is an advantage either way. Reusing from the lowest index on was just kind of how I reasoned about it in my head. I was thinking along the lines of keep index allocations ordered from 0 to N.